test/distributed/org/apache/cassandra/distributed/test/ring/BootstrapTest.java

```diff
@@ -95,7 +96,69 @@ public void autoBootstrapTest() throws Throwable
                 Assert.assertEquals("Node " + e.getKey() + " has incorrect row state", e.getValue().longValue(), 100L);
         }
     }
+    
+    @Test
+    public void hostIdOverrideTest() throws Throwable
+    {
+        int originalNodeCount = 2;
+        int expandedNodeCount = originalNodeCount + 1;
+        try (Cluster cluster = builder().withNodes(originalNodeCount)
+    									.withTokenSupplier(TokenSupplier.evenlyDistributedTokens(expandedNodeCount))
+    									.withNodeIdTopology(NetworkTopology.singleDcNetworkTopology(expandedNodeCount, "dc0", "rack0"))
+    									.withConfig(config -> config.with(NETWORK, GOSSIP))
+    									.start()) 
+    	{
+            populate(cluster,0, 100);
```

Comment:
@fibersel , do you use the correct formatter? I know it is a detail but there should be space after a comma (before "0").